### ISSUE TITLE
Add some way to display isocontours on an image in the plot

### ISSUE THREAD

**Issue by t20100** [silx-kit/silx#2242]
This has been requested.

**Comment by maurov** [comment]
**I vote for this feature**. Contour lines are very useful, for example, when dealing with Resonant Inelastic X-ray Scattering (RIXS) planes or raytracing simulations. Below a RIXS plane example with  `matplotlib.pyplot.imshow`  (left) and adding `matplotlib.pyplot.contour` on top (right). 

![image](https://user-images.githubusercontent.com/545294/48561409-cf6e4580-e8e7-11e8-8f7d-b9ae179bd709.png)

I am aware of `silx.examples.findContours`, but it would be very nice having something like a wrapper to `matplotlib.pyplot.contour` accessible in `PlotWidget`.

**Comment by vallsv** [comment]
How do you use `matplotlib.pyplot.contour`? You only uses it as a layer? Do you custom things?

I suppose using the colormap state (min/max linear/log) and displaying around 10/20 contours is enough to provide a good result for a first step. And easy to do.

Or you expect a user interface with a slider to pick a value for the isocontour. Or custom few things.

**Comment by maurov** [comment]
> How do you use `matplotlib.pyplot.contour`? You only uses it as a layer? Do you custom things?

Yes, I use `contour` as an additional layer to the image to permit the user better visualizing the (weak) features in the image. Nothing more.
 
> I suppose using the colormap state (min/max linear/log) and displaying around 10/20 contours is enough to provide a good result for a first step. And easy to do.

Yes, the user should be able to choose how many contours to plot and possibly the color and the line widths.
 
> Or you expect a user interface with a slider to pick a value for the isocontour. Or custom few things.

No interface is needed. I would propose to simply extend the API of `PlotWidget` to let the user decide if to show the contour on top of the image or not (by default), for example, this way:

`PlotWidget.addImage(... , contour=True, contour_levels=20, contour_colors='black', contour_linewidths=0.5)`

Here a complete minimal example showing the behaviour I have in mind:

```python
import numpy as np
from scipy.stats import multivariate_normal
import matplotlib.pyplot as plt
from matplotlib import cm
#
contour_levels = 20
contour_colors = 'gray'
contour_linewidths = 0.5
#
#dummy data
x = np.arange(-1, 1, 0.01)
y = np.arange(-1, 1, 0.01)
xx, yy = np.meshgrid(x, y)
zz0 = multivariate_normal.pdf(np.dstack((xx, yy)), mean=[0.5, -0.2], cov=[0.1, 0.05])
zz1 = multivariate_normal.pdf(np.dstack((xx, yy)), mean=[-0.25, 0.35], cov=[0.25, 0.35])
zz = zz0+zz1
#normalize colors from min to max
zzmin, zzmax = np.min(zz), np.max(zz)
norm = cm.colors.Normalize(vmin=zzmin, vmax=zzmax)
extent = (x.min(), x.max(), y.min(), y.max())
levels = np.linspace(zzmin, zzmax, contour_levels)
#
plt.close('all')
fig, axs = plt.subplots(1,2)
#no contours
axs[0].imshow(zz, origin='lower', extent=extent, cmap=cm.Reds, norm=norm)
#with contours
axs[1].imshow(zz, origin='lower', extent=extent, cmap=cm.Reds, norm=norm)
axs[1].contour(x, y, zz, levels, colors=contour_colors, linewidths=contour_linewidths)
#
plt.subplots_adjust()
plt.show()
```

![image](https://user-images.githubusercontent.com/545294/48617248-30eced80-e996-11e8-999c-ecc45217541c.png)

**Comment by vasole** [comment]
As far as we do not end up with something like this ...

![image](https://user-images.githubusercontent.com/5636251/48618274-25e78c80-e999-11e8-9a96-da5efa2e1bb7.png)

**Comment by t20100** [comment]
To me it was extending the plot item API (either by extending the image item or adding a new one) to enable configuring the contours and a small add-on in the `PlotWidget` to initialize it.
I would not go into configuration widgets.

**Comment by maurov** [comment]
@t20100 I vote for the _extension of the plot API_ to enable (+ minimum configuration) the overlay of the contour lines. The default behaviour will still be without contour, as it is now.

@vasole your example goes beyond just simply having the possibility to overlay the contour lines. How to find the contours in noisy data should be probably tackled with a specific widget, as is in `silx.examples.findContours`.
